SpatialGDK/Source/SpatialGDKServices/Private/SpatialGDKServicesModule.cpp

```diff
@@ -137,16 +139,37 @@ void FSpatialGDKServicesModule::ExecuteAndReadOutput(const FString& Executable,
 	FPlatformProcess::ClosePipe(0, WritePipe);
 }
 
+void FSpatialGDKServicesModule::SetProjectName(const FString& InProjectName)
+{
+	FString SpatialFileResult;
+
+	if (!JsonParsedSpatialFile)
+	{
+		UE_LOG(LogSpatialGDKServices, Error, TEXT("Failed to update project name(%s). Please ensure that the following file exists: %s"), *InProjectName, *SpatialGDKServicesConstants::SpatialOSConfigFileName);
+		return;
+	}
+	JsonParsedSpatialFile->SetStringField("name", InProjectName);
+
+	TSharedRef<TJsonWriter<>> JsonWriter = TJsonWriterFactory<>::Create(&SpatialFileResult);
+	if (!FJsonSerializer::Serialize(JsonParsedSpatialFile.ToSharedRef(), JsonWriter))
+	{
+		UE_LOG(LogSpatialGDKServices, Error, TEXT("Failed to write project name to parsed spatial file. Unable to serialize content to json file."));
```

Comment:
Link the `SpatialOSConfigFileName` here too, the file might be write access locked in P4 and users need to know how to unlock it.

Reply:
Wait no this one is not writing to file yet. This is for when it fails to serialize json into a string, so I don't know why this would ever happen. Writing to file is below, and we do print the filename then.